generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FlowSubscriptionExecutionStrategy.kt

```diff
@@ -103,21 +104,33 @@ class FlowSubscriptionExecutionStrategy(dfe: DataFetcherExceptionHandler) : Exec
             Let {fieldStream} be the result of running {ResolveFieldEventStream(subscriptionType, initialValue, rootField, argumentValues)}.
             Return {fieldStream}.
      */
+    @Suppress("UNCHECKED_CAST")
     private fun createSourceEventStream(
         executionContext: ExecutionContext,
         parameters: ExecutionStrategyParameters
     ): CompletableFuture<Flow<*>?> {
         val newParameters = firstFieldOfSubscriptionSelection(parameters)
 
         val fieldFetched = fetchField(executionContext, newParameters)
-        return fieldFetched.thenApply { fetchedValue ->
+        return if (fieldFetched is CompletableFuture<*>) {
```

Comment:
I think we need to rethink this logic and potentially update it to just match [SubscriptionExecutionStrategy](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/execution/SubscriptionExecutionStrategy.java#L93-L105), i.e. always wrap the fetched field in the future and keep the old logic as is?



- if it is a future -> we should handle it the same way as in the past (i.e. it should be publisher)
- if it is a flow ->

Reply:
in our case `Flow` (same as `Publisher`) will always be non CF value (i.e. I am unsure in what situation you would get a CF for subscription value).

Reply:
You might also want to check how `spring-graphql` is handling this (I'd assume they handle `Flow`)